src/main/resources/org/rust/lang/liveTemplates/Rust.xml

```diff
@@ -24,4 +24,142 @@
+        </context>
+    </template>
+
+    <template name="struct2" description="Struct with two fields" toReformat="true" toShortenFQNames="true"
+            value="struct $NAME$ {&#10;  $FIELD1$: $FIELD1TYPE$,&#10;  $FIELD2$: $FIELD2TYPE$,&#10;}$END$">
+        <variable name="NAME" expression="" defaultValue="" alwaysStopAt="true" />
+        <variable name="FIELD1" expression="" defaultValue="&quot;x&quot;" alwaysStopAt="true" />
+        <variable name="FIELD1TYPE" expression="" defaultValue="&quot;i32&quot;" alwaysStopAt="true" />
+        <variable name="FIELD2" expression="" defaultValue="&quot;y&quot;" alwaysStopAt="true" />
+        <variable name="FIELD2TYPE" expression="" defaultValue="&quot;i32&quot;" alwaysStopAt="true" />
+        <context>
+            <option name="RUST_FILE" value="true" />
+        </context>
+    </template>
+
+    <template name="tstruct1" description="Tuple struct with one field" toReformat="true" toShortenFQNames="true"
+            value="struct $NAME$($FIELD1TYPE$);$END$">
+        <variable name="NAME" expression="" defaultValue="" alwaysStopAt="true" />
+        <variable name="FIELD1TYPE" expression="" defaultValue="&quot;i32&quot;" alwaysStopAt="true" />
+        <context>
+            <option name="RUST_FILE" value="true" />
+        </context>
+    </template>
+
+    <template name="tstruct2" description="Tuple struct with two fields" toReformat="true" toShortenFQNames="true"
+            value="struct $NAME$($FIELD1TYPE$, $FIELD2TYPE$);$END$">
+        <variable name="NAME" expression="" defaultValue="" alwaysStopAt="true" />
+        <variable name="FIELD1TYPE" expression="" defaultValue="&quot;i32&quot;" alwaysStopAt="true" />
+        <variable name="FIELD2TYPE" expression="" defaultValue="&quot;i32&quot;" alwaysStopAt="true" />
+        <context>
+            <option name="RUST_FILE" value="true" />
+        </context>
+    </template>
+
+    <template name="enum" description="Enum" toReformat="true" toShortenFQNames="true"
+              value="enum $NAME$ {&#10;  $END$&#10;}">
+        <variable name="NAME" expression="" defaultValue="" alwaysStopAt="true" />
+        <context>
+            <option name="RUST_FILE" value="true" />
+        </context>
+    </template>
+
+    <template name="match" description="Match expression" toReformat="true" toShortenFQNames="true"
+              value="match $EXPRESSION$ {&#10;  $VAL1$ =&gt; $EXPRESSION1$,&#10;  _ =&gt; $EXPRESSION2$,&#10;}$END$">
+        <variable name="EXPRESSION" expression="" defaultValue="&quot;x&quot;" alwaysStopAt="true" />
+        <variable name="VAL1" expression="" defaultValue="&quot;true&quot;" alwaysStopAt="true" />
+        <variable name="EXPRESSION1" expression="" defaultValue="" alwaysStopAt="true" />
+        <variable name="EXPRESSION2" expression="" defaultValue="" alwaysStopAt="true" />
+        <context>
+            <option name="RUST_FILE" value="true" />
+        </context>
```

Comment:
I would suggest a more minimal

```
        match x {
            _ => $END$,
        }
```

The rationale is that the first match arm can be quite complicated, so it's better to exit template earlier. 

And once we have type inference and postfix templates, we would be able to do incredible stuff here!